compiler/parser/src/lexer.rs

```diff
@@ -241,59 +241,23 @@ where
 
     // Lexer helper functions:
     fn lex_identifier(&mut self) -> LexResult {
-        let mut name = String::new();
-        let start_pos = self.get_pos();
-
         // Detect potential string like rb'' b'' f'' u'' r''
-        let mut saw_b = false;
-        let mut saw_r = false;
-        let mut saw_u = false;
-        let mut saw_f = false;
```

Comment:
I think the current logic is a bit difficult to verify if it's correct.

Reply:
oh definitely, the change makes it much better.

Reply:
I think this originally came from the [tokenizer code](https://github.com/python/cpython/blob/3.11/Parser/tokenizer.c).